src/background/services/wallet/index.ts

```diff
@@ -1211,17 +1213,10 @@ export default class WalletService extends GenericService {
       await b.write();
     }
 
-    const newTXs = await this.exec('node', 'getTXByAddresses', addresses);
-
-    if (!newTXs.length) {
-      return transactions;
-    }
-
-    transactions = transactions.concat(newTXs);
-    return await this.getAllReceiveTXs(startDepth + ACCOUNT_DEPTH, endDepth + ACCOUNT_DEPTH, transactions);
+    return this.exec('node', 'getTXByAddresses', addresses, startDepth, endDepth);
   };
 
-  getAllChangeTXs = async (startDepth = 0, endDepth = ACCOUNT_DEPTH, transactions: any[] = []): Promise<any[]> => {
+  getAllChangeTXs = async (startDepth = 0, endDepth = 24): Promise<any[]> => {
```

Comment:
`24` seems a magical number that appears multiple times. Please consider replacing by a constant with meaningful name.